### DISCUSSION TITLE
(rust-analyzer) Global symbol picker does not show all symbols

### DISCUSSION THREAD

**Discussion by boxbeam** [helix-editor/helix#9046] [Troubleshooting]
Currently, the symbol picker for a file shows all symbols, including top-level functions, trait methods, and impl blocks. The global symbol picker shows only traits and types, and associated types in traits and impl blocks for some reason. It would be very helpful to at least be able to jump to functions from the global symbol picker.

**Comment by kirawi** [comment]
`rust-analyzer` limits the number of results sent to the client.  The list gets updated on filtering (but it's still capped).

**Comment by gabydd** [comment]
See https://github.com/helix-editor/helix/discussions/7867 and https://github.com/helix-editor/helix/discussions/8256 you need to tell rust-analyzer to send all symbols
